tests/test_library.py

```diff
@@ -0,0 +1,60 @@
+from attending import Library
+import pytest
+from pathlib import Path
+import os
+import shutil
+
+
+@pytest.fixture
+def ipython():
+    yield "ipython", "1.0.0", "https://readthedocs.org/projects/ipython/downloads/pdf/stable/"
+
+
+@pytest.fixture
+def lib_fixture():
+    base_path = Path(os.path.dirname(os.path.abspath(__file__))) / "test_dir"
+    yield base_path
+    if base_path.exists():
+        shutil.rmtree(base_path)
+
+
+def test_library_fetch(ipython, lib_fixture):
```

Comment:
weird! I had no idea `pytest` could do this kind of magic, where the `ipython` fixture from lines 8-10 get automagically added here, even though it's just a normal parameter.